### PR TITLE
deprecate templating component

### LINKED ISSUE #458: Deprecate the templating component integration
As of https://github.com/symfony/symfony/pull/51144, the `symfony/templating` component will be deprecated in Symfony 6.4 and removed in 7.0.  The integration in this bundle should be deprecated as well.

### PR DESCRIPTION
**Pull request by garak**
Fix #458

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

File: src/Provider/BuilderAliasProvider.php [hunk 1]
```diff
@@ -88,7 +88,7 @@ private function getBuilder(string $bundleName, string $className): object
             }
 
             $builder = new $try();
-            if ($builder instanceof ContainerAwareInterface) {
+            if (interface_exists(ContainerAwareInterface::class) && $builder instanceof ContainerAwareInterface) {
```
Comment (derrabus) [inline src/Provider/BuilderAliasProvider.php:91]: This change should not be necessary.

Comment (garak) [reply src/Provider/BuilderAliasProvider.php:91]: > This change should not be necessary.

Are you sure? The interface was removed, so with some Symfony versions doesn't exist.

Comment (derrabus) [reply src/Provider/BuilderAliasProvider.php:91]: Yes, but an `instanceof` on a non-existing class/interface is safe.

Comment (derrabus) [reply src/Provider/BuilderAliasProvider.php:91]: Revert the change and see for yourself. The tests should pass. 🙂

Comment (garak) [reply src/Provider/BuilderAliasProvider.php]: You were right. I reverted the change

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -59,9 +57,11 @@ jobs:
...
             - run: |
-                  sed -ri 's/"symfony\/(.+)": "(.+)"/"symfony\/\1": "'${{ matrix.symfony }}'"/' composer.json;
+                  sed -ri 's/"symfony\/(expression-language|framework-bundle|templating)": "(.+)"/"symfony\/\1": "'${{ matrix.symfony }}'"/' composer.json;
               if: matrix.symfony
             - run: composer config minimum-stability dev
               if: matrix.dev
             - run: composer update --no-interaction --no-progress --ansi ${{ matrix.composer_option }}
             - run: vendor/bin/phpunit
+              env:
+                  SYMFONY_DEPRECATIONS_HELPER: max[self]=1
```
Comment (stof) [inline .github/workflows/build.yaml:67]: This is a no-go. Tests covering our own deprecations should use `@group legacy` instead, to make deprecations allowed in them.

File: tests/Provider/BuilderAliasProviderTest.php [hunk 3]
```diff
@@ -5,10 +5,18 @@
...
 use PHPUnit\Framework\TestCase;
-use Symfony\Component\HttpKernel\Bundle\BundleInterface;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
 
 class BuilderAliasProviderTest extends TestCase
 {
+    protected function setUp(): void
+    {
+        parent::setUp();
+        if (!interface_exists(ContainerAwareInterface::class)) {
```
Comment (stof) [inline tests/Provider/BuilderAliasProviderTest.php:15]: how is this related to the templating component ?

Comment (stof) [reply tests/Provider/BuilderAliasProviderTest.php:15]: shouldn't we skip only the test relying on a container-aware menu and not all tests for BuilderAliasProvider ?

And anyway, this should not be in a commit or PR saying it deprecates the templating integration if its fixes a different problem.

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -59,9 +57,11 @@ jobs:
...
             - run: |
-                  sed -ri 's/"symfony\/(.+)": "(.+)"/"symfony\/\1": "'${{ matrix.symfony }}'"/' composer.json;
+                  sed -ri 's/"symfony\/(expression-language|framework-bundle|templating)": "(.+)"/"symfony\/\1": "'${{ matrix.symfony }}'"/' composer.json;
               if: matrix.symfony
             - run: composer config minimum-stability dev
               if: matrix.dev
             - run: composer update --no-interaction --no-progress --ansi ${{ matrix.composer_option }}
             - run: vendor/bin/phpunit
+              env:
+                  SYMFONY_DEPRECATIONS_HELPER: max[self]=0
```
Comment (stof) [inline .github/workflows/build.yaml:67]: This should be reverted. For other jobs that the `Lowest deps` one, we still want deprecations coming from vendors to fail the tests (outside legacy tests). The whole point of using the bridge is that our CI fail when we use deprecated APIs instead of discovering them only when our users complain they get deprecations in their projects.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (garak) [reply .github/workflows/build.yaml]: I added the needed groups to our legacy tests

File: tests/Provider/BuilderAliasProviderTest.php (hunk 3 above)
Comment (garak) [reply tests/Provider/BuilderAliasProviderTest.php]: You're right. I changed the skipping, now it skips only a single test

File: .github/workflows/build.yaml (hunk 4 above)
Comment (garak) [reply .github/workflows/build.yaml]: reverted

**Review by stof — APPROVED** [review]
